core/src/main/kotlin/fr/sncf/osrd/utils/JavaInteroperabilityTools.kt

```diff
@@ -2,16 +2,18 @@ package fr.sncf.osrd.utils
 
 import fr.sncf.osrd.sim_infra.api.Route
 import fr.sncf.osrd.sim_infra.api.RouteId
+import fr.sncf.osrd.sim_infra.api.SpeedLimitProperty
 import fr.sncf.osrd.utils.indexing.StaticIdxList
 import fr.sncf.osrd.utils.indexing.mutableStaticIdxArrayListOf
-import fr.sncf.osrd.utils.units.Speed
 
 /**
  * Java can't handle generics with value classes, it's impossible to extract the underlying speed
  * from a `RangeMapEntry` in pure java. This method handles this specific case.
  */
-fun rangeMapEntryToSpeed(entry: DistanceRangeMap.RangeMapEntry<Speed>): Long {
-    return entry.value.millimetersPerSecond.toLong()
+fun rangeMapEntryToSpeedLimitProperty(
```

Comment:
If handling java classes is too much of a pain, they can be automatically converted to kotlin at any point. We just haven't bothered converting classes that don't change much anymore.

Reply:
It would be nicer, but not mandatory here, so I let it like that.